docs/articles/clustering/cluster-sharding.md

```diff
@@ -257,3 +257,188 @@ In the normal operation of an Akka.NET cluster, the sharding system automaticall
 However, in the event that an `ActorSystem` is aborted as a result of a process / hardware failure it's possible that when using `akka.cluster.sharding.state-store-mode=persistence` leftover sharding data can still be present inside the Akka.Persistence journal and snapshot store - which will prevent the Akka.Cluster.Sharding system from recovering and starting up correctly the next time it's launched.
 
 This is a _rare_, but not impossible occurrence. In the event that this happens you'll need to purge the old Akka.Cluster.Sharding data before restarting the sharding system. You can purge this data automatically by [using the Akka.Cluster.Sharding.RepairTool](https://github.com/petabridge/Akka.Cluster.Sharding.RepairTool) produced by [Petabridge](https://petabridge.com/).
+
+## Tutorial
+
+In this tutorial, we will be making a very simple non-persisted shopping cart implementation using cluster sharding.
+All the code used in this tutorial can be found in the [GitHub repository](https://github.com/akkadotnet/akka.net/tree/dev/src/examples/Cluster/ClusterSharding/ShoppingCart)
+
+For a sample of a distributed data backed persistent example, please see [this example project](https://github.com/akkadotnet/akka.net/tree/dev/src/examples/Cluster/ClusterSharding/ClusterSharding.Node)
+instead.
+
+### Setting Up the Roles
+
+In a sharded cluster, it is common for the shards to be assigned their own specialized role inside the
+cluster to distribute their actors in. In this tutorial we will have a single frontend node that will
+feed three backend nodes with purchasing data. Usually, these nodes will be separated into different
+specialized projects but in this example, we will roll them into a single project and control their
+roles using an environment variable.
+
+```c#
```

Comment:
Need to use DocFx Code Snippets for samples https://dotnet.github.io/docfx/spec/docfx_flavored_markdown.html#code-snippet-sample

That way when code gets refactored, the samples on the website are automatically updated.

Reply:
Apply this to all code samples in this PR